src/test/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNodeTest.java

```diff
@@ -0,0 +1,155 @@
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import net.sf.saxon.Configuration;
+import net.sf.saxon.om.GenericTreeInfo;
+import net.sf.saxon.om.NodeInfo;
+
+public class AbstractNodeTest {
+
+    @Test
+    public void testCompareTo() {
+        final MockNode node1 = new MockNode(10, 1);
+        final MockNode node2 = new MockNode(8, 1);
+        assertWithMessage("Invalid comparison result")
+                .that(node1.compareOrder(node2))
+                .isEqualTo(2);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(-2);
+    }
+
+    @Test
+    public void testCompareToSameLines() {
+        final MockNode node1 = new MockNode(10, 12);
+        final MockNode node2 = new MockNode(10, 1);
+
+        assertWithMessage("Invalid comparison result")
+                .that(node1.compareOrder(node2))
+                .isEqualTo(11);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(-11);
+    }
+
+    @Test
+    public void testCompareToSameLinesAndColumns() {
+        final MockNode node1 = new MockNode(10, 12, "abc");
+        final MockNode node2 = new MockNode(10, 12, "def");
+
+        assertWithMessage("Invalid comparison result")
+                .that(node1.compareOrder(node2))
+                .isEqualTo(-3);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(3);
+    }
+
+    @Test
+    public void testCompareToIdentical() {
+        final MockNode node1 = new MockNode(10, 12, "abc");
+        final MockNode node2 = new MockNode(10, 12, "abc");
+
+        assertWithMessage("Invalid comparison result")
+                .that(node1.compareOrder(node2))
+                .isEqualTo(0);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(0);
+    }
+
+    /* default */ private static class MockNode extends AbstractNode {
```

Comment:
why default when we specifically define `private`? I thought `default` meant private-package.

Reply:
Good catch, probably make sense to make regexp check on this, to catch such copy paste in future.

Reply:
> probably make sense to make regexp check instance in our config on this, to catch such copy paste in future

moved to #7528